src/UserAdmin/src/Smart.FA.Catalog.Infrastructure/Services/S3StorageService.cs

```diff
@@ -65,4 +65,17 @@ public async Task UploadAsync(Stream fileStream, string fileName, CancellationTo
         StreamResponse? response = await _client.GetObjectAsync(request, cancellationToken);
         return response?.ResponseStream;
     }
+
+    /// <summary>
+    /// Get the pre-signed url
+    /// </summary>
+    /// <param name="fileName">File to fetch</param>
+    /// <param name="expirationDate">Date of expiration of the generated token</param>
+    /// <returns>The Stream of the file</returns>
+    public Uri? GetPreSignedUrl(string fileName, DateTime expirationDate)
+    {
+        var absoluteUrl =_client.GetPreSignedURL(new GetPreSignedUrlRequest {Key = fileName, BucketName = _options.ImageBucketName, Expires = expirationDate}) ?? null;
+        var uri = absoluteUrl is null ? null : new Uri(absoluteUrl);
+        return uri;
```

Comment:
You can return right away without creating a var
```suggestion
return absoluteUrl is null ? null : new Uri(absoluteUrl);
```